L/libjulia/common.jl

```diff
@@ -120,7 +121,7 @@ function configure(version)
     # julia expects libuv-julia.a
     override LIBUV=${prefix}/lib/libuv.a
 
-    override BB_TRIPLET_LIBGFORTRAN_CXXABI=${bb_full_target}
+    override BB_TRIPLET_LIBGFORTRAN_CXXABI=${bb_full_target%-julia_version*}
```

Comment:
Practically, this is fine, but theoretically there's no guarantee on the ordering of tags in a platform triplet.

I suggest we extend the `bb` command to allow the build script to ask for the platforms it wants.  E.g. `bb get-triplet --libgfortran --cxxabi`.

Reply:
(This is not a request on this PR, more a note that we should do this eventually)

Reply:
But isn't the precise value for`bb_full_target` what is used to generate the file names of the binary archives we upload to GitHub? And the precise value of `BB_TRIPLET_LIBGFORTRAN_CXXABI` (resp. the other `BB_TRIPLET*` values derived from it) what is then used to generate the download URLs? So they need to match?

What am I missing?

Reply:
No, the environment variable `${bb_full_target}` within the build environment has the full concrete specification of the target, including C++ string ABI and libgfortran version, even if those details are relevant for the generated tarballs

Reply:
I'm saying that your `bash` says "take `$bb_full_target` and remove everything after `-julia_version`, but if the triplet gets serialized as `x86_64-linux-gnu-julia_version+1.6.0-cxx11-libgfortran3`, then this obviously won't work.

Practically speaking, that's never going to happen because due to historical reasons we'll always put `libgfortran` and `cxx11` first, but some day in the future we may have other tags that are relevant here and they may or may not be placed before/after the `julia_version` tag.  So I'm saying it would be good for you as the builder to ask BB for a triplet containing a certain subset of tags.

Reply:
> No, the environment variable `${bb_full_target}` within the build environment has the full concrete specification of the target, including C++ string ABI and libgfortran version, even if those details aren't relevant for the generated tarballs

While that is true, the Julia build system is smart enough to strip off the `cxxabi` and `libgfortran` tags for dependencies that don't care about them.  The way Max is passing the target triplet in is correct, I am merely mentioning this for future-proofing.

Reply:
@giordano ah of course 🤦 that's what I was missing, thanks

Reply:
Yeah, I can of course change it to match something like `-julia_version[^-]+` or so (except that it'll have to fit into a bash regex).
But since this works now, I hope we can merge this as-is, and I promise I'll soon submit a cleaned version, to future-proof this. But then other things that are waiting for this PR could already proceed now.

Reply:
Actually I just fixed this anyway, `${bb_full_target/-julia_version+([^-])}` should do it